src/main/java/me/zeroeightsix/kami/module/modules/render/NoRender.kt

```diff
@@ -19,30 +25,67 @@ import net.minecraftforge.client.event.RenderBlockOverlayEvent
+    private val falling = register(Settings.booleanBuilder("Falling Blocks").withValue(true).withVisibility { page.value == Page.OTHER }.build())
+    val beacon = register(Settings.booleanBuilder("BeaconBeams").withValue(true).withVisibility { page.value == Page.OTHER }.build())
+    val skylight = register(Settings.booleanBuilder("SkyLightUpdates").withValue(true).withVisibility { page.value == Page.OTHER }.build())
+    private val enchantingTable = register(Settings.booleanBuilder("EnchantingBooks").withValue(true).withVisibility { page.value == Page.OTHER }.build())
+    private val enchantingTableSnow = register(Settings.booleanBuilder("EnchantBookSnow").withValue(false).withVisibility { page.value == Page.OTHER }.build())
+    private val projectiles = register(Settings.booleanBuilder("Projectiles etc").withValue(false).withVisibility { page.value == Page.OTHER }.build())
+    private val lightning = register(Settings.booleanBuilder("Lightning").withValue(true).withVisibility { page.value == Page.OTHER }.build())
+
+    //Generic
+    private val packets = register(Settings.booleanBuilder("Cancel Packets").withValue(true).withVisibility { page.value == Page.GENERIC }.build())
+
+    var entityList = HashSet<Class<*>>()
+
+    private enum class Page {
+        ENTITIES, OTHER, GENERIC
+    }
 
     init {
         listener<PacketEvent.Receive> {
-            if (it.packet is SPacketSpawnMob && mob.value ||
-                    it.packet is SPacketSpawnGlobalEntity && gEntity.value ||
-                    it.packet is SPacketSpawnObject && `object`.value ||
-                    it.packet is SPacketSpawnExperienceOrb && xp.value ||
+            if (it.packet is SPacketSpawnMob && mob.value && packets.value ||
+                    it.packet is SPacketSpawnGlobalEntity && lightning.value ||
+                    it.packet is SPacketSpawnExperienceOrb && xp.value && packets.value ||
                     it.packet is SPacketExplosion && explosion.value ||
-                    it.packet is SPacketSpawnPainting && paint.value ||
-                    it.packet is SPacketParticles && particles.value) it.cancel()
+                    it.packet is SPacketSpawnPainting && paint.value && packets.value ||
+                    it.packet is SPacketParticles && particles.value
+            ) it.cancel()
+            if (it.packet is SPacketSpawnObject) {
+                if (packets.value) {
+                    when (it.packet.type) {
+                        71 -> if (itemFrame.value) it.cancel()
+                        78 -> if (armorStand.value) it.cancel()
+                        51 -> if (crystal.value) it.cancel()
+                        2 -> if (items.value) it.cancel()
+                        70 -> if (falling.value) it.cancel()
+                    }
+                }
+                if (projectiles.value) it.cancel()
```

Comment:
Why is it cancelling packet without condition check here?

Reply:
that was a bug i am fixing it